src/components/navigation.tsx

```diff
@@ -0,0 +1,100 @@
+'use client';
+import {Disclosure} from '@headlessui/react';
+import {Bars3Icon, XMarkIcon} from '@heroicons/react/24/outline';
+import classNames from 'classnames';
+import {useUnlocalizedPathname} from 'next-intl/client';
+import Link from 'next/link';
+
+export default function Navigation({
```

Comment:
Would it be possible to define an `interface` for this? It's not required, but in my experience it improves the readability a little bit (less 'type noise' in the function signature).